pkg/provider/gcp/secretmanager/client.go

```diff
@@ -184,8 +189,14 @@ func (c *Client) PushSecret(ctx context.Context, payload []byte, metadata *apiex
 		}
 	}
 
+	unlock, ok := c.secretLocks.TryLock(secretName)
+	if !ok {
+		return fmt.Errorf("cannot hold a lock for the secret %s", secretName)
```

Comment:
From a UX perspective it shouldn't cause a NotReady. I guess we should retry acquiring the lock with a timeout (10s? 30s? 60s?). If we run into a timeout we can bail out an be `NotReady` :thinking:.

Reply:
Thanks for your feedback! I agree that we should handle the failure more gently. 

> I guess we should retry acquiring the lock with a timeout (10s? 30s? 60s?)

I'm not too sure if I understand this point correctly. Still, if that means the reconciliation should wait until the lock is released with a specific timeout, I'm a little concerned about the approach since that blocks other reconciliations. 

Instead, it might be more efficient if we just requeue the reconciliation request, and let the other reconciliations run while waiting for the lock to be released. The beauty of this approach is retry and backoff are all handled by the controller itself. I've updated the PR so, please let me know what you think on this point! 🙇 

- https://github.com/external-secrets/external-secrets/pull/2678/commits/a2cf39d32bc2718590aa5d97275b5d24218994a4